### DISCUSSION TITLE
The Correct Way to Setup HTTPS

### DISCUSSION THREAD

**Discussion by ilovefreesw** [WongSaang/chatgpt-ui#112] [General]
I am using Cloudflare and I have updated the IP of my VPS to my domain : mychat.example.com

I have enabled the flexible SSL mode.

Now, the HTTPS work. But on the homepage/chat page (mychat.example.com) only.

I can no longer access Admin panel at: mychat.example.com:9000.admin

It doesn't work. How would one fix that?

Also, please tell how to access the admin panel using the IP address of the VPS. In the docker-compose.yml file, I have added my domain name. So, when I access admin panel using IP address now, I get CSRF Failed error. Is there a way I can access the admin panel using both the domain and the server IP.

Thanks.

**Comment by WongSaang** [comment]
Hello, you can first look at these issues: https://github.com/WongSaang/chatgpt-ui/issues?q=is%3Aissue+CSRF+

**Comment by ilovefreesw** [comment]
Please read again. My issue is not related to CSRF errors explicitly. I know how to fix that. I just want to know how to make the admin panel accessible using IP address as well as domain.

Because. If it set APP_DOMAIN = mychat.example.com

then I can't access the admin panel using the IP address (12.45.67.89) of my server.

And If i set the APP_DOMAIN = 12.45.67.89

then I can't access the admin panel from my domain, mychat.xample.com:9000/admin.

How to set APP_DOMAIN to both and IP and domain so that can I access it from both.

**Comment by ParkerHorejs** [comment]
Hello! I'm a moron who managed to get the UI up and running on an ubuntu server. I want to set up SSL using certificates I've already created for my domain (certbot), but I don't know how. What is this flexible SSL module you speak of, and can you give me any pointers? Thanks in advance!

**Reply by ilovefreesw** [reply]
> Hello! I'm a moron who managed to get the UI up and running on an ubuntu server. I want to set up SSL using certificates I've already created for my domain (certbot), but I don't know how. What is this flexible SSL module you speak of, and can you give me any pointers? Thanks in advance!

Flexible SSL is not the same as SSL certificates. It's just if you create it in Cloudflare then it will appear to your visitors that they are using HTTPS.

In Cloudflare, create a A record (without proxied) and set its value to the IP address of your server.
GO to Rules and then create a new configuration rule for the hostname. Add your subdomain there and in the action select Add SSL and set the mode to Flexible.

This works for me.

**Reply by ParkerHorejs** [reply]
Thanks! I actually managed to get it working by using a reverse proxy in my VirtualHosts with apache. I got that CSRF issue when attempting to access the admin panel by raw IP instead of the actual domain - I'm wondering whether it limits access to the domain you specify when deploying the containers.

**Reply by ilovefreesw** [reply]
> Thanks! I actually managed to get it working 

I think you set APP_DOMAIN to domain name. Set it to the IP address of the server instead, that way you will be able to access the admin panel.
